pkg/actions/cluster/unowned.go

```diff
@@ -21,3 +36,198 @@ func (c *UnownedCluster) Upgrade(dryRun bool) error {
 	_, err := upgrade(c.cfg, c.ctl, dryRun)
 	return err
 }
+
+func (c *UnownedCluster) Delete(waitTimeout time.Duration, wait bool) error {
+	clusterName := c.cfg.Metadata.Name
+
+	if err := c.checkClusterExists(clusterName); err != nil {
+		return err
+	}
+
+	clientSet, err := c.ctl.NewStdClientSet(c.cfg)
+	if err != nil {
+		return err
+	}
+
+	if err := deleteSharedResources(c.cfg, c.ctl, clientSet, waitTimeout); err != nil {
+		return err
+	}
+
+	// we have to wait for nodegroups to delete before deleting the cluster
+	// so the `wait` value is ignored here
```

Comment:
what is the `wait` value?

Reply:
@michaelbeaumont we have the `wait` bool to decide if the command should block until deletion is complete or not https://github.com/weaveworks/eksctl/pull/2969/files#diff-fa7f0dfa823f7f99001cd3d411d89673edec1d379968dda30880f7d13153feceR40

Reply:
Ah right, missed that somehow